src/methods/join.js

```diff
@@ -162,9 +201,26 @@ function joinResults(realm: Realm, joinCondition: AbstractValue,
   if (result1 instanceof AbruptCompletion && result2 instanceof AbruptCompletion) {
     return new JoinedAbruptCompletions(realm, joinCondition, result1, e1, result2, e2);
   }
-  if (result1 instanceof Value && result2 instanceof Value)
+  if (result1 instanceof Value && result2 instanceof Value) {
     return joinValues(realm, result1, result2, getAbstractValue);
-  return new PossiblyNormalCompletion(joinCondition, result1, e1, result2, e2);
```

Comment:
The constructor fails unless one of the constituent completions is an Abrupt completion and that has not yet been established at this point, hence the subsequent code.